src/core/drive/navigator.ts

```diff
@@ -97,7 +97,12 @@ export class Navigator {
           shouldCacheSnapshot,
           response: { statusCode, responseHTML, redirected },
         }
-        this.proposeVisit(fetchResponse.location, visitOptions)
+
+        const location = fetchResponse.location
+        if (redirected) {
+          location.hash = formSubmission.fetchRequest.location.hash
```

Comment:
> When a form submission redirects to a URL containing a hash
>
> -- description from https://github.com/hotwired/turbo/issues/825

In my understanding, the "redirects to a URL" part is describing a URL decided on the server-side. This implementation seems like it's reading the `formSubmission.fetchRequest.location.hash` value from a URL decided on the client-side.

To make it concrete, I was hoping to highlight the fact that Turbo cannot access the `anchor:` value from `redirect_to posts_url(anchor: helpers.dom_id(@post))`, and not the `#a-hash` value from the `[action]` attribute from an element like `<form action="/posts#a-hash"></form>`.

Reply:
With that being said, let's fix both!

I think the server-side value will be inaccessible for the same reasons that headers in intermediate redirect responses are inaccessible (see the "Making it work with Turbo Frames and redirects" heading in https://github.com/hotwired/turbo/issues/257#issuecomment-968131438).

Reply:
any idea how we could make Turbo know about hashes coming from the server? It looks like we have two bugs here and this implementation fixes the one I mentioned

Reply:
@seanpdoyle looks like there's no way to get the hash coming from the server since fetch responses are always stripped https://github.com/node-fetch/node-fetch/issues/211#issuecomment-265353508

Would it be ok to move forward with the client fix first?

Reply:
I just found https://github.com/whatwg/fetch/issues/1167 as a proposal to expose the fragment for the response url. I mentioned the Turbo tickets I found about it as use cases for that proposal.